### PR TITLE
[BACKLOG-40223] Move Hadoop UI (Named Cluster Dialog) dialog from OSGi to standard kettle plugin

### PR DESCRIPTION
**Pull request by e-cuellar**

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>




---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+  public static boolean processDriverFile( String driverFile, HadoopClusterManager manager ) throws Exception {
+    boolean result = false;
+    File file = new File( driverFile );
+    FileInputStream driverStream = new FileInputStream( file );
+    if ( isValidUpload( file.getName(), FileType.DRIVER, manager ) ) {
+      String destination = Const.getShimDriverDeploymentLocation();
+      String fileSeparator = System.getProperty( "file.separator" );
+      FileUtils.copyInputStreamToFile( driverStream,
+        new File( destination + fileSeparator + file.getName() ) );
```
Comment (buildguy) [inline kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java:192]: [comment]: <> (FrogbotReviewComment)


## 🎯 Static Application Security Testing (SAST) Vulnerability
<div align='center'>

| Severity                | Finding                  |
| :---------------------: | :-----------------------------------: |
| <br>  Medium | Stored Path Traversal |

</div>

<details>
<summary> <b>Full description</b> </summary>
<br>


### Overview
Path traversal, also known as directory traversal, is a type of
vulnerability that allows an attacker to access files or directories on a
computer or device that are outside of the intended directory.
Allowing arbitrary read access can allow the attacker to read sensitive
files, such as configuration files or sensitive data, potentially leading
data loss or even system compromise. Allowing arbitrary write access is
more severe and in most cases leads to arbitrary code execution, via
editing important system files or sensitive data.

### Vulnerable example
```java
public class path_traversaLvuln {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        File docFile = docPath.toFile();
        FileUtils.copyFile(docFile, response.getOutputStream());
    }
}
```
In this example, an attacker can, via a stored parameter, inject a back-path,
that will get anywhere in the system, using "../../".

### Remediation
```diff
public class path_traversal_safe {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        + Path normDocPath = docPath.normalize();
        + // Make sure the canonical path resides in the desired dir
        + if (normDocPath.startsWith(DOCS_FOLDER)) {
            File docFile = docPath.toFile();
            FileUtils.copyFile(docFile, response.getOutputStream());
        + }
    }
}
```
By checking that the folder name still starts with the predefined prefix, we
make sure that the attacker is not able to back-path outside of the allowed
folder.


</details>

<details>
<summary> <b>Code Flows</b> </summary>
<br>


<details>
<summary> <b>Vulnerable data flow analysis result</b> </summary>
<br>


↘️ `System.getProperty( "file.separator" )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `destination + fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 192)

↘️ `destination + fileSeparator + file.getName()` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 192)

↘️ `new File( destination + fileSeparator + file.getName() )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 192)


</details>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

Comment (buildguy) [inline kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java:192]: [comment]: <> (FrogbotReviewComment)


## 🎯 Static Application Security Testing (SAST) Vulnerability
<div align='center'>

| Severity                | Finding                  |
| :---------------------: | :-----------------------------------: |
| <br>  Medium | Stored Path Traversal |

</div>

<details>
<summary> <b>Full description</b> </summary>
<br>


### Overview
Path traversal, also known as directory traversal, is a type of
vulnerability that allows an attacker to access files or directories on a
computer or device that are outside of the intended directory.
Allowing arbitrary read access can allow the attacker to read sensitive
files, such as configuration files or sensitive data, potentially leading
data loss or even system compromise. Allowing arbitrary write access is
more severe and in most cases leads to arbitrary code execution, via
editing important system files or sensitive data.

### Vulnerable example
```java
public class path_traversaLvuln {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        File docFile = docPath.toFile();
        FileUtils.copyFile(docFile, response.getOutputStream());
    }
}
```
In this example, an attacker can, via a stored parameter, inject a back-path,
that will get anywhere in the system, using "../../".

### Remediation
```diff
public class path_traversal_safe {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        + Path normDocPath = docPath.normalize();
        + // Make sure the canonical path resides in the desired dir
        + if (normDocPath.startsWith(DOCS_FOLDER)) {
            File docFile = docPath.toFile();
            FileUtils.copyFile(docFile, response.getOutputStream());
        + }
    }
}
```
By checking that the folder name still starts with the predefined prefix, we
make sure that the attacker is not able to back-path outside of the allowed
folder.


</details>

<details>
<summary> <b>Code Flows</b> </summary>
<br>


<details>
<summary> <b>Vulnerable data flow analysis result</b> </summary>
<br>


↘️ `System.getProperty( "file.separator" )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `destination + fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 192)

↘️ `destination + fileSeparator + file.getName()` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 192)

↘️ `new File( destination + fileSeparator + file.getName() )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 192)


</details>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 49m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **1277**, Failures: **0**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2629/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>




---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    boolean result = false;
+    File file = new File( driverFile );
+    FileInputStream driverStream = new FileInputStream( file );
+    if ( isValidUpload( file.getName(), FileType.DRIVER, manager ) ) {
+      String destination = Const.getShimDriverDeploymentLocation();
+      String fileSeparator = System.getProperty( "file.separator" );
+      String driverDestination = destination + fileSeparator + file.getName();
+      if ( driverDestination.startsWith( destination ) ) {
+        FileUtils.copyInputStreamToFile( driverStream,
+          new File( driverDestination ) );
```
Comment (buildguy) [inline kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java:194]: [comment]: <> (FrogbotReviewComment)


## 🎯 Static Application Security Testing (SAST) Vulnerability
<div align='center'>

| Severity                | Finding                  |
| :---------------------: | :-----------------------------------: |
| <br>  Medium | Stored Path Traversal |

</div>

<details>
<summary> <b>Full description</b> </summary>
<br>


### Overview
Path traversal, also known as directory traversal, is a type of
vulnerability that allows an attacker to access files or directories on a
computer or device that are outside of the intended directory.
Allowing arbitrary read access can allow the attacker to read sensitive
files, such as configuration files or sensitive data, potentially leading
data loss or even system compromise. Allowing arbitrary write access is
more severe and in most cases leads to arbitrary code execution, via
editing important system files or sensitive data.

### Vulnerable example
```java
public class path_traversaLvuln {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        File docFile = docPath.toFile();
        FileUtils.copyFile(docFile, response.getOutputStream());
    }
}
```
In this example, an attacker can, via a stored parameter, inject a back-path,
that will get anywhere in the system, using "../../".

### Remediation
```diff
public class path_traversal_safe {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        + Path normDocPath = docPath.normalize();
        + // Make sure the canonical path resides in the desired dir
        + if (normDocPath.startsWith(DOCS_FOLDER)) {
            File docFile = docPath.toFile();
            FileUtils.copyFile(docFile, response.getOutputStream());
        + }
    }
}
```
By checking that the folder name still starts with the predefined prefix, we
make sure that the attacker is not able to back-path outside of the allowed
folder.


</details>

<details>
<summary> <b>Code Flows</b> </summary>
<br>


<details>
<summary> <b>Vulnerable data flow analysis result</b> </summary>
<br>


↘️ `System.getProperty( "file.separator" )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `destination + fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 191)

↘️ `destination + fileSeparator + file.getName()` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 191)

↘️ `driverDestination` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 191)

↘️ `new File( driverDestination )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 194)


</details>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>




---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java (hunk 2 above)
Comment (buildguy) [inline kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java:194]: [comment]: <> (FrogbotReviewComment)


## 🎯 Static Application Security Testing (SAST) Vulnerability
<div align='center'>

| Severity                | Finding                  |
| :---------------------: | :-----------------------------------: |
| <br>  Medium | Stored Path Traversal |

</div>

<details>
<summary> <b>Full description</b> </summary>
<br>


### Overview
Path traversal, also known as directory traversal, is a type of
vulnerability that allows an attacker to access files or directories on a
computer or device that are outside of the intended directory.
Allowing arbitrary read access can allow the attacker to read sensitive
files, such as configuration files or sensitive data, potentially leading
data loss or even system compromise. Allowing arbitrary write access is
more severe and in most cases leads to arbitrary code execution, via
editing important system files or sensitive data.

### Vulnerable example
```java
public class path_traversaLvuln {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        File docFile = docPath.toFile();
        FileUtils.copyFile(docFile, response.getOutputStream());
    }
}
```
In this example, an attacker can, via a stored parameter, inject a back-path,
that will get anywhere in the system, using "../../".

### Remediation
```diff
public class path_traversal_safe {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String DOCS_FOLDER = "/srv/www/docs";
        String docName = statement.executeQuery(query);  // Reading from DB
        Path docPath = Paths.get(DOCS_FOLDER, docName);
        + Path normDocPath = docPath.normalize();
        + // Make sure the canonical path resides in the desired dir
        + if (normDocPath.startsWith(DOCS_FOLDER)) {
            File docFile = docPath.toFile();
            FileUtils.copyFile(docFile, response.getOutputStream());
        + }
    }
}
```
By checking that the folder name still starts with the predefined prefix, we
make sure that the attacker is not able to back-path outside of the allowed
folder.


</details>

<details>
<summary> <b>Code Flows</b> </summary>
<br>


<details>
<summary> <b>Vulnerable data flow analysis result</b> </summary>
<br>


↘️ `System.getProperty( "file.separator" )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 190)

↘️ `destination + fileSeparator` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 191)

↘️ `destination + fileSeparator + file.getName()` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 191)

↘️ `driverDestination` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 191)

↘️ `new File( driverDestination )` (at kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java line 194)


</details>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 46m 35s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **1277**, Failures: **0**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2629/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java (hunk 2 above)
Comment (e-cuellar) [reply kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/util/NamedClusterHelper.java]: Fixed

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 46s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **1277**, Failures: **0**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2629/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by angel-ramoscardona — APPROVED** [review]

File: kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/NamedClusterDialog.java [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  public boolean isDevMode() {
+    return isDevMode;
+  }
+
+  public boolean isEditMode() {
+    return isEditMode;
+  }
+
+  /*
+  public static void main( String[] args ) {
```
Comment (rmansoor) [inline kettle-plugins/hadoop-cluster/ui/src/main/java/org/pentaho/big/data/kettle/plugins/hadoopcluster/ui/dialog/wizard/NamedClusterDialog.java:343]: @e-cuellar Can we remove this code ? Do you need this for some reason ?

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Pull request closed without merging**